### PR TITLE
implement #162

### PR DESCRIPTION
**Pull request by marcmengel**
Push chunks of dagnabbit parser state on stack; also keep list of start and end of each parallel branch, since serial within parallel has a different end than start job for each line.  Also clean up condor_submit_dag wrapper a bit.

### CODE REVIEW

**Comment by marcmengel** [comment]
Also added fixes for #163, mainly small template changes.

**Comment by marcmengel** [comment]
Also including a fix for the discovered bug that we were looking for the pushed proxy in /tmp rather than the current directory in DAGs, which is  unhappy.   We could get rid of that check by copying the proxy into the submit sandbox and just always make it local, but it's a smaller change to make the template smarter...

**Comment by marcmengel** [comment]
Okay, so now we take:
```
jobsub -n --mail_on_error $SUBMIT_FLAGS  ${JOBSUB_EXPORTS} file://./jobA.sh
<parallel>
jobsub -n --mail_on_error $SUBMIT_FLAGS  ${JOBSUB_EXPORTS} file://./jobB.sh
<serial>
jobsub -n --mail_on_error $SUBMIT_FLAGS  ${JOBSUB_EXPORTS} file://./jobC.sh
jobsub -n --mail_on_error $SUBMIT_FLAGS  ${JOBSUB_EXPORTS} file://./jobD.sh
jobsub -n --mail_on_error $SUBMIT_FLAGS  ${JOBSUB_EXPORTS} file://./jobE.sh
</serial>
</parallel>
jobsub -n --mail_on_error $SUBMIT_FLAGS  ${JOBSUB_EXPORTS} file://./jobF.sh
</serial>
```
and turn it into 
```
DOT dag.dot UPDATE
JOB stage_1 stage_1.cmd
VARS stage_1 JOBSUBJOBSECTION="1" nodename="$(JOB)"
JOB stage_2 stage_2.cmd
VARS stage_2 JOBSUBJOBSECTION="2" nodename="$(JOB)"
JOB stage_3 stage_3.cmd
VARS stage_3 JOBSUBJOBSECTION="3" nodename="$(JOB)"
JOB stage_4 stage_4.cmd
VARS stage_4 JOBSUBJOBSECTION="4" nodename="$(JOB)"
PARENT stage_3 CHILD stage_4
JOB stage_5 stage_5.cmd
VARS stage_5 JOBSUBJOBSECTION="5" nodename="$(JOB)"
PARENT stage_4 CHILD stage_5
PARENT stage_1 CHILD stage_2 stage_3
JOB stage_6 stage_6.cmd
VARS stage_6 JOBSUBJOBSECTION="6" nodename="$(JOB)"
PARENT stage_2 stage_5 CHILD stage_6
```
which  runs and makes the pretty picture:

dagTest6

File: bin/jobsub_submit [hunk 1]
```diff
@@ -193,10 +193,11 @@ def main():
     #    shutil.copyfile(token, token_dest)
     #    varg["token"] = token_dest
 
+    varg["is_dag"] = True
```
Comment (shreyb) [inline bin/jobsub_submit:196]: I understand why you're setting `varg["is_dag"]` this way, but I worry that six months from now, we'll see this line and wonder why we've "defaulted" to setting `varg["is_dag"]` to `True`.  Would it be possible, if slightly less elegant to do something along the lines of:

```python
varg["is_dag"] = False
if args.dag or args.dataset_definition or args.maxConcurrent:
    varg["is_dag"] = True
```
or even 

```python
varg["is_dag"] = True if (args.dag or args.dataset_definition or args.maxConcurrent) else False
```

And then having the `if` block on 197?

I could even see having line 196 be 
```python
varg["is_dag"] = False
```
and then setting `varg["is_dag"]` to `True` for every applicable clause in the `if` statement starting in line 197.  

I know this is extra code compared to what you wrote here, I just worry about the future understandability of this bit.

**Review by shreyb — CHANGES_REQUESTED** [review]
Looks good, except for the one comment in `bin/jobsub_submit`.

File: bin/jobsub_submit (hunk 1 above)
Comment (shreyb) [reply bin/jobsub_submit:196]: Yeah - it's certainly _nicer_ the way you did it here, but I do have concerns about the clarity.

**Review by shreyb — APPROVED** [review]
Looks good - thanks!

**Comment by marcmengel** [comment]
Yes, both.

**Pull request merged**